app/routes/meta/versions.py

```diff
@@ -174,6 +176,23 @@ async def update_version(
     row: ORMVersion = await versions.update_version(dataset, version, **input_data)
     # TODO: Need to clarify routine for when source_uri has changed. Append/ overwrite
 
+    if "source_uri" in input_data:
+        if row.is_mutable:
+            # append
+            version_data: ORMVersion = await versions.get_version(dataset, version)
+            version_data.update(input_data)
```

Comment:
Do you try to update the db record, or only the in memory object?

Reply:
just in memory, I changed to just add fields we need to input_data